### PR TITLE
Fix wrong casing in suggestions at start of sentence

### PR DESCRIPTION
**Pull request by Fatalll**

### CODE REVIEW

File: src/main/kotlin/tanvd/grazi/grammar/GrammarChecker.kt [hunk 1]
```diff
@@ -88,6 +88,16 @@ class GrammarChecker(private val ignoreChar: LinkedSet<(CharSequence, Char) -> B
                     val newRange = IntRange(typo.location.range.start + startShift - range.start, typo.location.range.endInclusive + endShift - range.start)
                     typo.copy(location = typo.location.copy(range = newRange, pointer = firstToken.toPointer()))
                 } else null
+            }?.let { typo ->
```
Comment (TanVD) [inline src/main/kotlin/tanvd/grazi/grammar/GrammarChecker.kt]: Ok, it is definitely should not be here. Grammar checker is an abstract checker that works for all languages. And here is a specific fix for specific rule of English. Think about moving it to some global `LangToolFixes` that will provide fixes for language-tool lang rules that we consider broken.

Comment (Fatalll) [reply src/main/kotlin/tanvd/grazi/grammar/GrammarChecker.kt]: Done

File: src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package tanvd.grazi.grammar
+
+import org.languagetool.rules.Rule
+
+enum class LangToolFixes(private val ruleId: String, private val fix: (suggestion: String) -> String) {
+    ARTICLE_MISSING("ARTICLE_MISSING", { suggestion ->
+        when {
+            suggestion.startsWith("The") -> "The " + suggestion[4].toLowerCase() + suggestion.substring(5)
```
Comment (TanVD) [inline src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt]: What if it has more that one whitspace?

File: src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            else -> suggestion
+        }
+    });
+
+    companion object {
+        fun fixSuggestion(ruleId: String, suggestion: String): String {
+            return values().find { it.ruleId == ruleId }?.let { it.fix(suggestion) } ?: suggestion
+        }
+
+        fun fixSuggestion(rule: Rule, suggestion: String): String {
```
Comment (TanVD) [inline src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt]: Use `=` when you have only one return

File: src/main/kotlin/tanvd/grazi/grammar/Typo.kt [hunk 4]
```diff
@@ -30,8 +29,9 @@ data class Typo(val location: Location, val info: Info, val fixes: List<String>
 
     constructor(match: RuleMatch, lang: Lang, offset: Int = 0) : this(
             Location(match.toIntRange().withOffset(offset)),
-            Info(lang, match.rule, match, match.typoCategory), match.suggestedReplacements)
-
+            Info(lang, match.rule, match, match.typoCategory),
+            match.suggestedReplacements.map { LangToolFixes.fixSuggestion(match.rule, it) }
```
Comment (TanVD) [inline src/main/kotlin/tanvd/grazi/grammar/Typo.kt:57]: What if I will use main constructor? Move replace of suggestedReplacements to `init`

File: src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package tanvd.grazi.grammar
```
Comment (TanVD) [inline src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt]: Move it to `tanvd.grazi.language` package

File: src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt (hunk 2 above)
Comment (Fatalll) [reply src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt]: There is no possibility to get more than one whitespace, btw added regex to replace it

File: src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt (hunk 3 above)
Comment (Fatalll) [reply src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt]: Fix

File: src/main/kotlin/tanvd/grazi/grammar/Typo.kt (hunk 4 above)
Comment (Fatalll) [reply src/main/kotlin/tanvd/grazi/grammar/Typo.kt:57]: For LangTool intended only this constructor, so added a comment

File: src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt (hunk 5 above)
Comment (Fatalll) [reply src/main/kotlin/tanvd/grazi/grammar/LangToolFixes.kt]: Done

**Review by TanVD — APPROVED** [review]

**Pull request merged**